dashboard/app/models/user.rb

```diff
@@ -1646,6 +1646,39 @@ def associate_with_potential_pd_enrollments
     end
   end
 
+  # When creating an account, we want to look for any channels that got created
+  # for this user before they signed in, and if any of them are in our Applab HOC
+  # course, we will create a UserScript entry so that they get a course card
+  def generate_progress_from_storage_id(storage_id, script_name='applab-intro')
+    # applab-intro is not seeded in our minimal test env used on test/circle. We
+    # should be able to handle this gracefully
+    begin
+      script = Script.get_from_cache(script_name)
+    rescue ActiveRecord::RecordNotFound
+      nil
+    end
+    return unless script
```

Comment:
I'm not sure if this is the best way to handle this. I was a little surprised to find that `Script.get_from_cache(script_name)` threw instead of returning `nil`. I could make it so that we include applab-intro in the set of things we seed for tests, but it seems like this method really ought to be resilient to it not existing anyways.